iam/snippets/src/test/java/ServiceAccountTests.java

```diff
@@ -78,139 +76,295 @@ public void tearDown() {
-
-    assertTrue(response.iterateAll().iterator().hasNext());
+  public void testServiceAccount_listServiceAccounts() throws IOException, InterruptedException {
+    // Prepare
+    String serviceAccountName = generateServiceAccountName();
+    try {
+      Util.setUpTest_createServiceAccount(PROJECT_ID, serviceAccountName);
+
+      // Act
+      IAMClient.ListServiceAccountsPagedResponse response =
+          ListServiceAccounts.listServiceAccounts(PROJECT_ID);
+
+      // Assert
+      assertTrue(response.iterateAll().iterator().hasNext());
+    } finally {
+      // Cleanup
+      Util.tearDownTest_deleteServiceAccount(PROJECT_ID, serviceAccountName);
+    }
   }
 
   @Test
-  public void stage2_testServiceAccountRename() throws IOException {
-    String renameTo = "your-new-display-name";
-    ServiceAccount serviceAccount = RenameServiceAccount
-            .renameServiceAccount(PROJECT_ID, SERVICE_ACCOUNT, renameTo);
-    String got = bout.toString();
-    assertThat(got, containsString("Updated display name"));
-    assertThat(got, containsString(renameTo));
-    assertNotNull(serviceAccount);
-    assertThat(renameTo, containsString(serviceAccount.getDisplayName()));
+  public void testServiceAccount_getServiceAccount() throws IOException, InterruptedException {
+    // Prepare
+    String serviceAccountName = generateServiceAccountName();
+    try {
+      Util.setUpTest_createServiceAccount(PROJECT_ID, serviceAccountName);
+
+      // Act
+      ServiceAccount account = GetServiceAccount.getServiceAccount(PROJECT_ID, serviceAccountName);
+
+      // Assert
+      assertTrue(account.getName().contains(serviceAccountName));
+      assertEquals(PROJECT_ID, account.getProjectId());
+    } finally {
+      // Cleanup
+      Util.tearDownTest_deleteServiceAccount(PROJECT_ID, serviceAccountName);
+    }
   }
 
   @Test
-  public void stage2_testServiceAccountGet() throws IOException {
-    ServiceAccount account = GetServiceAccount.getServiceAccount(PROJECT_ID, SERVICE_ACCOUNT);
-
-    assertTrue(account.getName().contains(SERVICE_ACCOUNT));
-    assertEquals(PROJECT_ID, account.getProjectId());
+  public void testServiceAccount_renameServiceAccount() throws IOException, InterruptedException {
+    // Prepare
+    String serviceAccountName = generateServiceAccountName();
+    String newServiceAccountName = "new-" + generateServiceAccountName();
+    try {
+      Util.setUpTest_createServiceAccount(PROJECT_ID, serviceAccountName);
+
+      // Act
+      ServiceAccount renamedServiceAccount =
+          RenameServiceAccount.renameServiceAccount(
+              PROJECT_ID, serviceAccountName, newServiceAccountName);
+
+      // Assert
+      String got = bout.toString();
+      assertThat(got, containsString("Updated display name"));
+      assertThat(got, containsString(newServiceAccountName));
+      assertNotNull(renamedServiceAccount);
+      assertThat(newServiceAccountName, containsString(renamedServiceAccount.getDisplayName()));
```

Comment:
nit: IFAC the name SA name should match and not be "contained". Consider using more strict assertion.